src/Stratis.Bitcoin/Consensus/ConsensusExceptions.cs

```diff
@@ -28,6 +28,30 @@ public ConnectHeaderException() : base()
         }
     }
 
+    /// <summary>
+    /// This throws when the header of a previously block that failed
+    /// partial or full validation and was marked as invalid is passed to the node.
+    /// </summary>
+    public class HeaderInvalidException : ConsensusException
+    {
+        public HeaderInvalidException() : base()
+        {
+        }
+    }
+
+    /// <summary>
+    /// An exception that is contains exception coming from the <see cref="IConsensusRuleEngine"/> execution.
+    /// </summary>
+    public class ConsensusRuleException : ConsensusException
```

Comment:
this is basically a copy-paste of ConsensusErrorException

Reply:
LOL yes it is even the name is wrong they should be swapped but didnt want to make such a big change yet.